### PR TITLE
Fixes initializing random on windows

### PR DESCRIPTION
**Pull request by cammm**
The SENTRY_CTOR isn't working for me and afaik it is not safe to call LoadLibraryA outside of main() ie during static initialization or DLLMain. Switching this to initialize on the first call to random instead.

### CODE REVIEW

File: src/sentry_random.c [hunk 1]
```diff
@@ -54,19 +54,20 @@ getrandom_devurandom(void *dst, size_t bytes)
 #ifdef SENTRY_PLATFORM_WINDOWS
 typedef BOOLEAN(WINAPI *sRtlGenRandom)(PVOID Buffer, ULONG BufferLength);
 
-static sRtlGenRandom pRtlGenRandom;
-
-SENTRY_CTOR (init_winapi) {
```
Comment (Swatinem) [inline src/sentry_random.c]: I think this was the only usage of `SENTRY_CTOR`, was it? In which case would be good to also remove it.

Comment (cammm) [reply src/sentry_random.c]: Yeah good idea, removed

**Comment by jan-auer** [comment]
@Swatinem what do you think about creating a random number in `sentry_init` just to eager-initialize and avoid overheads when the first event is created?

**Comment by Swatinem** [comment]
We basically do that already, when creating a random uuid for the run folder.

**Pull request merged**